### ISSUE TITLE
Resource Mapping - Support groupings and namespace/FQN

### ISSUE THREAD

**Issue by ttschampel** [opentdf/platform#666]
As a publisher and consumer of resource mappings I'd like to:
-  bundle groups of Resource Mappings into a set `ResourceMappingSet`
-  reference the ResourceMappingSet by FQN
- enforce access and share ResourceMappingSets by namespace
- CRUD ResourceMappingSets
- Find ResourceMappingSets/ResourceMappings by FQN, Namespace

**Comment by ttschampel** [comment]
The addressable/usability concerns similar to #585 , but for resource mappings

**Comment by jakedoublev** [comment]
Hey @ttschampel, some follow-up questions for you:

Right now, [resource mappings are associated with attribute values](https://github.com/opentdf/platform/blob/main/service/policy/db/migrations/20240212000000_schema_erd.md). Do you envision RM Sets to group only RMs on the same attribute definition? In other words, an example RM Set could be an ANY_OF attribute with 4 values, each value containing an RM, and the 4 RMs together equating to the RM Set? 

Regarding FQN lookup, do you mean we should have an FQN for a Resource Mapping, or that you should be able to look up an RM by its associated attribute value FQN? In that case, it seems like we'd just need to make sure the FQN lookup exposes all mappings for the relevant value being looked up?

Lastly, as RMs are currently only associated with attribute values, and not definitions or namespaces, is there a use case I am not aware of for mapping RMs to namespaces? Alternatively, is it more about the query so that given a namespace you can find all RMs associated with values on attributes under that namespace?

**Comment by ttschampel** [comment]
still required

**Comment by damorris25** [comment]
Yes, I think this needs an ADR type discussion. I'd like to address this gap in some way in an upcoming release in the near future.
